### PR TITLE
Plot tree update 

### PR DESCRIPTION
**Pull request by MichaelSev**
Plot Tree function

- Additions to issue https://github.com/ComputationalEvolutionaryMorphometry/hyperiax/issues/4
- Changed the slanted lines to vertical and horizontal setup
- Included names on tree
- Included edge length on vertical  lines
- Fixed a problem of overlapping leaf node on leaf with different depths
- moved plot_2d to a specific plot_utils.py file

Updated nootbook examples to include the new plotting

Updated builder.asymmetric_tree to the new parent/child setup

### CODE REVIEW

File: hyperiax/tree/builders.py [hunk 1]
```diff
@@ -37,28 +36,22 @@ def asymmetric_tree(h: int):
...
 
     for _ in range(h - 1):
-        node.children = [TreeNode(),TreeNode()]
-    
-        node.children[0].parent = node
-        node.children[1].parent = node
-
-        node = node.children[1]
+        node.children = ChildList()
+        node.children=  [TreeNode(children=ChildList(),parent=node), TreeNode(children=ChildList(),parent=node)]
```
Comment (frozaken) [inline hyperiax/tree/builders.py:49]: I think this should be `ChildList([TreeNode(children=ChildList(),parent=node), TreeNode(children=ChildList(),parent=node)])`, right?

File: hyperiax/tree/builders.py [hunk 2]
```diff
@@ -37,28 +36,22 @@ def asymmetric_tree(h: int):
...
-    root = TreeNode(children=[TreeNode(),TreeNode()])
-
-    root.children[0].parent = root
-    root.children[1].parent = root
+        return HypTree(ChildList([TreeNode()]))
 
-    node = root.children[1]
+    # Fake root 
+    root = TreeNode(); root.parent = None; root.children = ChildList()
+    root.children = [TreeNode(children=ChildList(),parent=root), TreeNode(children=ChildList(),parent=root)]
```
Comment (frozaken) [inline hyperiax/tree/builders.py:43]: same thing here

File: hyperiax/tree/tree.py [hunk 3]
```diff
@@ -97,6 +97,20 @@ def iter_bfs(self):
...
+
+        Yields:
+            iterator: An iterator that runs over the nodes.
+        """
+        stack = deque([self.root])
+
+        while stack:
+            current = stack.pop()
+            if current.children:
+                stack.extend(reversed(current.children))
```
Comment (frozaken) [inline hyperiax/tree/tree.py:111]: Is there a reason for the reversal?

File: hyperiax/tree/tree.py [hunk 4]
```diff
@@ -114,68 +128,28 @@ def iter_levels(self):
             if children := queue.popleft().children:
                 buffer_queue.extend(children)
 
+    def iter_leaves_dfs(self):
```
Comment (frozaken) [inline hyperiax/tree/tree.py:131]: Is a leaf iteration dfs not the same as a leaf iteration bfs?

**Review by frozaken — CHANGES_REQUESTED** [review]

File: hyperiax/tree/builders.py (hunk 1 above)
Comment (MichaelSev) [reply hyperiax/tree/builders.py:49]: I agree

File: hyperiax/tree/builders.py (hunk 2 above)
Comment (MichaelSev) [reply hyperiax/tree/builders.py:43]: I agree

**Pull request closed without merging**